src/daft-schema/src/schema.rs

```diff
@@ -334,3 +334,17 @@ impl TryFrom<&arrow2::datatypes::Schema> for Schema {
         Self::new(daft_fields)
     }
 }
+
+/// Custom impl of PartialEq because IndexMap PartialEq does not check for ordering
+impl PartialEq for Schema {
+    fn eq(&self, other: &Self) -> bool {
+        self.fields.len() == other.fields.len()
+            && self
+                .fields
+                .iter()
+                .zip(other.fields.iter())
+                .all(|(s, o)| s == o)
+    }
+}
+
+impl Eq for Schema {}
```

Comment:
Are these changes to `schema.rs` related?

Reply:
see my comment in monotonically increasing id!